packages/listbox/test/lion-listbox.test.js

```diff
@@ -0,0 +1,284 @@
+      await el.updateComplete;
+      expect(el.querySelectorAll('lion-option')[1].focused).to.be.true;
+    });
+
+    it('navigates to top and bottom with [home] [end] keys', async () => {
+      const el = await fixture(html`
+        <lion-listbox>
+          <lion-option value="nr1">Item 1</lion-option>
+          <lion-option value="nr2">Item 2</lion-option>
+          <lion-option value="nr3">Item 3</lion-option>
+          <lion-option value="nr4">Item 4</lion-option>
+        </lion-listbox>
+      `);
+      el.dispatchEvent(new KeyboardEvent('keydown', { key: 'Home' }));
+      await el.updateComplete;
+      expect(el.querySelectorAll('lion-option')[0].focused).to.be.true;
+
+      el.dispatchEvent(new KeyboardEvent('keydown', { key: 'End' }));
+      await el.updateComplete;
+      expect(el.querySelectorAll('lion-option')[3].focused).to.be.true;
+    });
+
+    it('doesn\t "rotate" option selection on [up/down] arrow keys', async () => {
+      const el = await fixture(html`
+        <lion-listbox>
+          <lion-option selected value="nr1">Item 1</lion-option>
+          <lion-option value="nr2">Item 2</lion-option>
+        </lion-listbox>
+      `);
+      keyUpOn(el, keyCodes.arrowDown);
+      expect(el.querySelectorAll('lion-option')[1].selected).to.be.true;
+      expect(el.value).to.equal('nr2');
+
+      keyUpOn(el, keyCodes.arrowDown);
+      expect(el.querySelectorAll('lion-option')[1].selected).to.be.true;
+      expect(el.value).to.equal('nr2');
+    });
+
+    it('selects an option with [enter] key', async () => {
+      const el = await fixture(html`
+        <lion-listbox>
+          <lion-option value="nr1">Item 1</lion-option>
+          <lion-option value="nr2">Item 2</lion-option>
+          <lion-option value="nr3">Item 3</lion-option>
+          <lion-option value="nr4">Item 4</lion-option>
+        </lion-listbox>
+      `);
+      el.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
+      await el.updateComplete;
+      expect(el.querySelectorAll('lion-option')[3]).to.be.equal('inline-block');
+
+      el.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
+      await el.updateComplete;
+      expect(el.querySelector('[slot="content"]').style.display).to.be.equal('none');
+    });
+
+    // TODO: nice to have
+    it.skip('switch focus with single [character] key', async () => {
+      const el = await fixture(html`
+        <lion-listbox>
+          <lion-option value="aaa">Aaa</lion-option>
+          <lion-option value="bbb">Bbb</lion-option>
+          <lion-option value="ccc">Ccc</lion-option>
+        </lion-listbox>
+      `);
+      el.querySelectorAll('lion-option')[0].focus();
+      el.dispatchEvent(new KeyboardEvent('keydown', { key: 'C' }));
+      await el.updateComplete;
+      expect(el.querySelectorAll('lion-option')[2].focused).to.be.true;
+    });
+
+    it('switch focus to specific element with multiple [character] keys', async () => {
```

Comment:
In my opinion, as soon as it goes further than the 1st character key toggling through options that start with this character, it becomes a search, and should be its own component, probably part of auto-complete. This is also not a feature of the default WAI-ARIA listbox it seems.

Which reminds me, have we thought about how an auto-complete would work structurally? Will it be an extension of LionRichSelect where the invoker is overridden to have some extra behavior? I guess the ListboxBehavior will also be overridden to filter out options that don't match the search?

I know that's not part of the MVP but it makes sense to think about how it would be done before we set in stone the foundations.

Reply:
The WAI-ARIA listbox example has this feature. It is usefull if you have a listbox with country names for instance.

For the autocomplete it will be the same as the `lion-rich-select` only with an input instead of a button. The listbox options will indeed by filtered out dynamically.

Reply:
Ahh I see now, there's a timeout that resets it to the first char. I think we want that too then, and we should have a test for it :). Nothing more annoying than mistyping the first char and not being able to go back